Apalis imx6 secondary boot

igoropaniuk opened the pull request:
Please do not submit a Pull Request via github.  Our project makes use of
mailing lists for patch submission and review.  For more details please
see https://www.denx.de/wiki/U-Boot/Patches

igoropaniuk committed:
[FIO internal] imx: add support of secondary boot for MX6

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

igoropaniuk committed:
[FIO internal] fastboot: fsl: add sit offset for MX6 SoCs

Add Secondary Image Table offset for all MX6 SoCs, which is used
in redundant boot setups.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

igoropaniuk commented on board/toradex/apalis_imx6/apalis_imx6.c:
```diff
@@ -1040,6 +1040,27 @@ static void spl_dram_init(void)
 
+#if defined(CONFIG_SECONDARY_BOOT_RUNTIME_DETECTION) && \
+    defined(CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_USE_SECTOR)
+unsigned long spl_mmc_get_uboot_raw_sector(struct mmc *mmc,
```
As another option we can even move that part somwhere to `arch/arm/mach-imx/` to avoid code duplication (as now we have `spl_mmc_get_uboot_raw_sector()` in `boards/freescale/common/mmc.c` and another one here `spl_mmc_get_uboot_raw_sector`)

igoropaniuk replied:
@ricardosalveti @mike-scott

ricardosalveti replied:
Yeah, moving to a more generic place might be better to avoid duplication. We want all of these changes to be generic to the respective imx soc, so makes sense to make it generic code-wise.

igoropaniuk replied:
fixed, please check updated version

ricardosalveti reviewed (commented):
Just one comment regarding the code location, but LGTM otherwise.

igoropaniuk committed:
[FIO internal] imx: move secondary boot logic to soc-specific code

Move spl_mmc_get_uboot_raw_sector to SoC-specific sources instead of
board files.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

ricardosalveti reviewed (approved):
LGTM

mike-scott reviewed (approved):
LGTM
